### PR TITLE
Final Code for Review

### PR DESCRIPTION
**Pull request by Dineshg49**

### CODE REVIEW

File: experiment/simulation/css/main.css [hunk 1]
```diff
@@ -1 +1,24 @@
...
+	overflow-y: scroll;
+}
+
+.main {
+	width: 85%;
+	height: 85%;
+	background-color: white;
+}
+
+.action-button {
```
Comment (raj-vlabs) [inline experiment/simulation/css/main.css]: If this class is not used anywhere, please remove it

File: experiment/simulation/js/main.js [hunk 2]
```diff
@@ -1 +1,293 @@
...
+		document.getElementById("stress").innerHTML = stress[step].toString() + " Mpa";
+		document.getElementById("strain").innerHTML = strain[step].toString();
+		document.getElementById("load").innerHTML = load[step].toString() + " N";
+		document.getElementById("deformation").innerHTML = deformation[step].toString() + " mm";
+		if(step<load.length)
+		{
+			chart.render();
+			step++;
+		}
+		if(step == load.length)
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js]: Please use the === operator unless the == is specifically needed

File: experiment/simulation/js/main.js [hunk 3]
```diff
@@ -1 +1,293 @@
...
+		{
+			drawStatic();
+			drawObject(ctx,brokenBoxPart1, boxColor);
+			drawObject(ctx,brokenBoxPart2, boxColor);
+			window.clearTimeout(tmHandle);
+		}
+		else
+		{
+			tmHandle = 	window.setTimeout(draw, 5000 / fps);
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js]: This code will execute when step<load.length
Is this the intended outcome?

**Review by raj-vlabs — CHANGES_REQUESTED** [review]

File: experiment/simulation/js/main.js (hunk 3 above)
Comment (Dineshg49) [reply experiment/simulation/js/main.js]: yes sir it is the intended outcome.

File: experiment/references.md [hunk 4]
```diff
@@ -1 +1 @@
-### Link your references in here
\ No newline at end of file
+- https://en.wikipedia.org/wiki/Compressive_strength
```
Comment (raj-vlabs) [inline experiment/references.md]: Please use meaningful text for the link description

File: experiment/simulation/index.html [hunk 5]
```diff
@@ -1,13 +1,80 @@
 <!DOCTYPE html>
 <html>
+
 <head>
-	<!-- Add CSS at the head of HTML file -->
-    <link rel="stylesheet" href="./css/main.css">
+	<meta charset="utf-8">
+	<meta name="viewport" content="width=device-width, initial-scale=1">
+	<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.1/css/bulma.min.css">
```
Comment (raj-vlabs) [inline experiment/simulation/index.html]: Remove Bulma import if you are using virtual-style. If something breaks, please talk to Shlok to include that in the virtual-style library

**Review by raj-vlabs — CHANGES_REQUESTED** [review]

File: experiment/simulation/index.html (hunk 5 above)
Comment (Dineshg49) [reply experiment/simulation/index.html]: yes sir some of the components broke, I would contact shlok regarding this.

File: experiment/procedure.md [hunk 6]
```diff
@@ -1 +1,13 @@
-### Procedure
\ No newline at end of file
+ 
+**Procedure**
+
+1. Dimensions of test piece is measured at 3 different places along its height/length to determine the average c/s area.
+2. Ends of the specimen should be plane. For that the ends are tested on a bearing plate.
```
Comment (raj-vlabs) [inline experiment/procedure.md]: The spelling of plane here is different from the spelling in posttest question 1 (plain). Please use the correct spelling at both places.

File: experiment/procedure.md [hunk 7]
```diff
@@ -1 +1,13 @@
...
+3. The specimen is placed centrally between the two compression plates, such that the center of moving head is vertically above the center of specimen.
+4. Load is applied on the specimen by moving the movable head.
+5. The load and the corresponding contraction are measured at different intervals. The load interval may be as 500kg.
+6. Load is applied until the specimen fails.
+7. The values of stress and strain are calculated at several intervals and are plotted on a graph.
+8. The resultant graph is the Stress Strain curve which is the property of a material and is independent of the dimensions of specimen used.
```
Comment (raj-vlabs) [inline experiment/procedure.md]: Please use a hyphen - between Stress and Strain to make it look like the graph is plotted between these 2 parameters.

File: experiment/simulation/index.html [hunk 8]
```diff
@@ -1,13 +1,79 @@
...
+            <div>
+                <div id="chartContainer"></div>
+            </div>
+        </div>
+    </div>
+</div>
+
+<script src="https://unpkg.com/mathjs@9.2.0/lib/browser/math.js"></script>
+<script src="https://canvasjs.com/assets/script/canvasjs.min.js"></script>
+<script src='https://cdn.plot.ly/plotly-latest.min.js'></script>
```
Comment (raj-vlabs) [inline experiment/simulation/index.html]: Why have plotly and mathjs been imported here?

**Review by raj-vlabs — CHANGES_REQUESTED** [review]

File: experiment/simulation/js/main.js [hunk 9]
```diff
@@ -1 +1,272 @@
...
+    const rodWidth = 240;
+    const upperRodHeight = 300;
+    const lowerRodHeight = 360;
+    const boxWidth = 320;
+    const boxStarty = 500;
+    const breakLength = 80;
+    const gap = 10;
+    const benchHeight = 100;
+
+    const boxColor = "#2C9AD1"; // virtual labs color
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js]: Please take out the color codes and put them in a separate file with mnemonic keys. Look here for example:
https://github.com/virtual-labs/exp-compaction-test-iiith/blob/main/experiment/simulation/js/data.json

**Review by raj-vlabs — CHANGES_REQUESTED** [review]

**Review by raj-vlabs — APPROVED** [review]

**Pull request merged**